src/Knp/Menu/MenuItem.php

```diff
@@ -93,35 +93,36 @@ class MenuItem implements ItemInterface
     /**
      * Class constructor
      *
-     * @param string           $name    The name of this menu, which is how its parent will
+     * @param string|null      $name    The name of this menu, which is how its parent will
      *                                  reference it. Also used as label if label not specified
      * @param FactoryInterface $factory
      */
-    public function __construct($name, FactoryInterface $factory)
+    public function __construct(?string $name, FactoryInterface $factory)
```

Comment:
Do we really want to allow `null` here? Is wasn't allowed before.

Reply:
Indeed it was allowed. Of course, we can change such behavior.

Reply:
A menu item needs a name (it's in an invalid state if it doesn't have one). So I don't see the point of allowing null here?

Reply:
> Indeed it was allowed.

Yes, but it was cast to an empty string. You've removed that cast.

Reply:
@wouterj current class has a default value of null on property, that mislead me. I'll change to accept strings only

Reply:
> @wouterj current class has a default value of null on property, that mislead me.

We can remove the `null` initialization from all properties, if they confuse you. They're redundant anyway.

Reply:
It looks like a test is currently testing for a `null` name, see https://github.com/KnpLabs/KnpMenu/blob/master/tests/Knp/Menu/Tests/MenuItemGetterSetterTest.php#L27
So, is it a legitimate case?

Reply:
To recap: the old code was allowing for `null` value, but then was casting it to a string (an empty string, of course).
Do we want to keep it? Or we prefer to disallow empty names?

Reply:
So the options are:
* Make the parameter nullable and restore the string cast.
* Disallow `null` completely, adjust the test and force application code to pass a string instead.

Because of @wouterj's comment, I'd prefer the second option, but I could live with the first option as well.

Reply:
I'm also for the second option. Let see what Wouter thinks about it

Reply:
disallowing null is a BC break. So we need to deprecate it. We cannot just change it.

Reply:
@stof this PR is all about BC breaks, it's meant to be released in a major version (e.g. `3.0.0`)

Reply:
but you should not do BC break for which there is no deprecation in the previous major version, as that breaks the upgrade path for users (they would get hit by the BC break, instead of getting a warning before the upgrade to prepare).

Reply:
@stof I just published release `2.5.0` where we add such deprecation